indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -1066,7 +1075,8 @@ private synchronized void persistSequences() throws IOException
                 ingestionState,
                 getTaskCompletionUnparseableEvents(),
                 getTaskCompletionRowStats(),
-                errorMsg
+                errorMsg,
+                errorMsg == null
```

Comment:
I think we should add a comment here that says something to the effect of "For streaming ingestion, segments are considered to be available immediately if the task is successful. If the task failed, the segments are not available"

At least that's what I think this is trying to say.

Reply:
yes, it is saying that a successful streaming task means segments were confirmed to be handed off. an unsuccessful task means we are not positive that segments were handed off. I'm unsure if it is possible to have a streaming task confirm that segments are handed off and still fail, but I assume it is not very common.